hdfs_namenode/datadog_checks/hdfs_namenode/hdfs_namenode.py

```diff
@@ -187,3 +194,16 @@ def _join_url_dir(cls, url, *args):
             url = urljoin(url, path.lstrip('/'))
 
         return url
+
+    def _collect_metadata(self, value):
+        # only get first info block
+        data = next(iter(value))
```

Comment:
Any chance the beans value could be empty (in the wild), leading this to raise `StopIteration`? If so, maybe consider using a default:

```python
data = next(iter(value), {})
```

(I'm not very familiar with HDFS or JMX, maybe there's an external spec that guarantees `value` always contains at least one data block?)